### PR TITLE
Add token refresh logic

### PR DESCRIPTION
**Pull request by chentom88**
Signed-off-by: Rachel Heaton <rheaton@pivotal.io>

*Note: Please remember to review the Datadog [Contribution Guidelines](https://github.com/pivotal-cloudops/dd-agent/blob/master/CONTRIBUTING.md)
if you have not yet done so.*

### What does this PR do?

This change will cause the nozzle to refresh it's UAA token when it expires

### Motivation

Without this change, the nozzle will only fetch a UAA token when it starts up and then will fail continuously after it expires

### Testing Guidelines

An overview on [testing](https://github.com/pivotal-cloudops/dd-agent/blob/master/tests/README.md)
is available in our contribution guidelines.

### CODE REVIEW

File: internal/client/cloudfoundry/loggregator_client.go [hunk 1]
```diff
@@ -42,12 +73,14 @@ func newRLPGatewayClientDoer(token string, insecureSkipVerify bool) *rlpGatewayC
 	}
 
 	return &rlpGatewayClientDoer{
-		token:  token,
-		client: client,
+		token:        "",
```
Comment (jamespollard8) [inline internal/client/cloudfoundry/loggregator_client.go]: hmm, it's not clear to me why the token changed to `""`

Comment (chentom88) [reply internal/client/cloudfoundry/loggregator_client.go]: This is just to be explicit about setting the string to it's zero value. This line is not specifically necessary.

File: internal/client/cloudfoundry/loggregator_client.go [hunk 2]
```diff
@@ -16,23 +17,53 @@ import (
...
+	}
+
 	req.Header.Set("Authorization", d.token)
-	return d.client.Do(req)
+	resp, err := d.client.Do(req)
+	if err != nil {
+		return resp, err
+	}
+
+	// If token is bad, try to refresh token and retry request once but only if disable access
```
Comment (xtreme-bozhidar-lenchov) [inline internal/client/cloudfoundry/loggregator_client.go]: Hope I'm not misreading the comment, but it looks like we attempt to refresh the token and retry the request in the block below if `disableACS` is false. Is the last part of the comment incorrect then (`retry request once but only if disable access control is enabled`)?

Comment (chentom88) [reply internal/client/cloudfoundry/loggregator_client.go]: You are reading that correct, if disableACS is false then access control is enabled. In which case, we do want to use tokens. We only want to skip token nonsense if access control is disabled. It's unfortunate that the flag is worded the way it is.

**Pull request merged**